Feat/review paging 기능 구현

shCHO9801 opened the pull request:
### 변경사항
ISSUE : #40
**AS-IS**
- 모든 리뷰 및 사용자 본인의 리뷰 데이터를 한 번에 반환
- 정렬/필터 기능 없음
- 공통 페이지 응답 형식 없음

**TO-BE**
- Pageable 기반 페이지네이션 처리
- 전체 리뷰: 최신순/과거순 정렬, 카테고리 필터링 가능
- 내 리뷰: 기본 페이지네이션 지원
- 응답 형식 통일 (공통 PagedResponseDto 사용)

### 테스트
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [x] API 테스트 
- ```/api/reviews```
![스크린샷 2025-04-04 오전 3 29 00](https://github.com/user-attachments/assets/2ad9b5bd-91ec-4340-8dee-acd5958a1bf5)

- ```/api/reviews/my```
![스크린샷 2025-04-04 오전 3 29 22](https://github.com/user-attachments/assets/f6031839-b653-4405-9c8a-fc309d3297dd)

shCHO9801 committed:
feat : job-roles 를 반환하는 endpoint 추가

shCHO9801 committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feat/review-paging

shCHO9801 committed:
feat : Review 조회 기능이 Paging Response 로 반환되도록 구현

geonuu09 commented on boottalk/src/main/java/com/icandoit/boottalk/review/controller/ReviewController.java:
```diff
@@ -34,20 +38,35 @@ public ResponseEntity<ReviewResponseDto> create(@RequestBody @Valid ReviewCreate
-	public ResponseEntity<List<ReviewResponseDto>> listAll() {
-		// TODO : 페이징 처리
-		return ResponseEntity.ok(reviewService.listAll());
+	public ResponseEntity<PagedResponseDto<ReviewResponseDto>> listAll(
```
성현님, listAll()이라는 메소드명이 다소 범용적이라 어떤 리뷰를 조회하는 메소드인지 한눈에 알기 어려울 것 같습니다. 예를 들어, 특정 조건이 있는지(예: 최신 리뷰, 특정 사용자 리뷰) 명확하게 드러나도록 listAllReviews() 또는 getAllReviews() 같은 방식으로 표현하면 더 직관적일 것 같아요! 😊

shCHO9801 replied:
넵 해당 내용 수정했습니다!

geonuu09 reviewed (approved):
고생하셨습니다!

P-Taeyoung commented:
고생하셨습니다~

shCHO9801 committed:
refactor : Transactional(readOnly) 어노테이션 삭제, 메소드 명들 명시적으로 표현

shCHO9801 committed:
Merge remote-tracking branch 'origin/develop' into feat/review-paging

shCHO9801 committed:
test: 리뷰 서비스 테스트 수정 및 추가

- 리뷰 생성 시 Course 평점 및 리뷰 수 증가 테스트 수정
- 리뷰 생성 실패 테스트 추가 (중복 리뷰 작성 시 예외)
- 리뷰 수정 실패 테스트 추가 (리뷰 없음 시 예외)
- 리뷰 삭제 실패 테스트 추가 (작성자 불일치 시 예외)
- 부트캠프 리뷰 조회 실패 테스트 추가 (부트캠프 없음 시 예외)
